### PR TITLE
Strip all variation selectors on emoji

### PR DESCRIPTION
**Pull request by dbkr**
...when inserting into or looking up in the unicode to emoji map.

This broke with emojibase 4.2.0 which changed the type of a whole
load of emojis to 'text' when previously they were 'emoji'. This
caused them to get the 'text' variant of the unicode string which
has the text variation selector (15) appended instead of the emoji
variation selector (16). We were only stripping the emoji selector,
so upgrading to 4.2.0 caused riot to fail to find the heart in the
unicode map, which therefore prevented the app from starting.

### CODE REVIEW

File: src/emoji.js [hunk 1]
```diff
@@ -80,3 +77,23 @@ EMOJIBASE.forEach(emoji => {
...
+function stripVariation(str) {
+    let ret = '';
+    for (let i = 0; i < str.length; ++i) {
+        const charCode = str.charCodeAt(i);
+        // append to output only if it's outside the variation selector range
+        if (charCode < 0xFE00 && charCode > 0xFE0F) {
+            ret += str.charAt(i);
+        }
+    }
+    return ret;
```
Comment (t3chguy) [inline src/emoji.js]: can this not just be done using a regex unicode character range?

Comment (dbkr) [reply src/emoji.js]: Good question: I'm not sure as I'd have to look at whether js regexes match by UTF-16 character code or by codepoint, which I suspect depends on whether you use an old style regex or an ES6 style 'u' regex?

Comment (dbkr) [reply src/emoji.js]: nm: a regex is indeed fine. My code wasn't though because I had an 'and' when I needed an 'or'.

File: src/emoji.js [hunk 2]
```diff
@@ -80,3 +77,15 @@ EMOJIBASE.forEach(emoji => {
...
     }
 });
+
+/**
+ * Strips variation selectors from a string
+ * NB. Skin tone modifers are not variation selectors:
+ * this function does not touch them. (Should it?)
+ *
+ * @param {string} str string to strip
+ * @returns {string} stripped string
```
Comment (t3chguy) [inline src/emoji.js]: Does that work without the character class [...]

Comment (dbkr) [reply src/emoji.js]: bleh, no, and also it doesn't work unless you use a regex literal. Not sure how I managed to mis-test it...

**Pull request merged**